scripts/Course.lua

```diff
@@ -88,6 +88,12 @@ function Course:getFieldPolygon()
 	return self.fieldPolygon
 end
 
+--- Any vine nodes found on the field?
+function Course:hasVineNodesOnTheField()
+	local x, _, z = self:getWaypointPosition(2)
```

Comment:
Magic constants are rarely good and they certainly need an explanation.

Reply:
Your right this solution is a suboptimal solution, as this assumes the second waypoint is at least on the field.